dot/rpc/modules/author_test.go

```diff
@@ -33,10 +33,7 @@ func TestAuthorModule_HasSessionKey_WhenScaleDataEmptyOrNil(t *testing.T) {
 		decodeSessionKeysMock.ReturnArguments = []interface{}{dec, err}
 	})
 
-	module := &AuthorModule{
-		coreAPI: coremockapi,
-		logger:  log.New("service", "RPC", "module", "author"),
-	}
+	module := NewAuthorModule(log.New("service", "RPC", "module", "author"), coremockapi, nil)
```

Comment:
Do we care about the logs in the test here? Maybe change NewAuthorModule to accept the `log15.Logger` interface and use a mock expecting anything instead (or your own no-op test logger implementation)?

Reply:
@qdm12 I'm a little confused on this one, aren't we going to be replacing the log15 library?

Reply:
Yes we are, it's just that I see sooo much logs in my tests, I'm trying to restrict them at least for new tests. Or at least make people aware, we shouldn't need logs by default in our tests :wink: 

What you could do, which is easier, is:

```go
logger := log.New("service", "RPC", "module", "author")
handler := log.StreamHandler(ioutil.Discard, log.TerminalFormat())
handler = log.CallerFileHandler(handler)
logger.SetHandler(log.LvlFilterHandler(log.LvlTrace, handler))

module := NewAuthorModule(logger, coremockapi, ...)
```

You can write a little test helper function to create a logger that discards everything by default too, if you want to re-use it across tests. You can have that helper function take a boolean to enable actual logging (swap `ioutil.Discard` with `os.Stdout`), when we need to debug a test with additional logs.